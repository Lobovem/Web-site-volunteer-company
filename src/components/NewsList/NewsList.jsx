import { Link } from 'react-router-dom';
import { TitleMain } from '../TitleMain/TitleMain';
import s from './NewsList.module.scss';

export const NewsList = ({ currentItems }) => {
  return (
    <div className={s.news}>
      <h3 className={s.news__decor}>Новини</h3>

      <div className={s.news__wrap}>
        {currentItems &&
          currentItems.map((item) => (
            <div className={s.news__inner} key={item.id}>
              <div className={s.news__imgWrap}>
                <Link key={item.id} to={`/news/${item.id}`}>
                  <img className={s.news__img} src={item.photoMain} alt="news.jpg" />
                </Link>
              </div>

              <div className={s.news__textBox}>
                <div className={s.news__textWrap}>
                  <Link className={s.news__linkAlt} to={`/news/${item.id}`}>
                    <TitleMain title={item.title}></TitleMain>
                  </Link>

                  <p className={s.news__desc}>{item.desc}</p>
                </div>

                <Link to={`/news/${item.id}`} className={s.news__link}>
                  Читати повністю
                </Link>
              </div>
            </div>
          ))}
      </div>
    </div>
  );
};
