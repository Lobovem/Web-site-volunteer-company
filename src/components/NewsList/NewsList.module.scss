@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.news {
  position: relative;

  &__decor {
    margin: 0;
    padding: 0;
    display: none;

    @media (min-width: 1500px) {

      position: absolute;
      display: block;
      top: 150px;
      left: -430px;
      z-index: 0;
      font-weight: $font-weight-500;
      font-size: $font-size-165;
      line-height: 165px;
      text-align: center;
      text-transform: uppercase;
      color: $color-grey;
      mix-blend-mode: normal;
      transform: rotate(-90deg);
    }
  }

  &__wrap {
    display: flex;
    gap: 40px;
    flex-direction: column;


    @media (min-width:750px) {

      div:nth-child(odd) div {
        order: 0;
      }
    }
  }

  &__inner {
    display: flex;
    flex-direction: column;
    align-items: center;

    @media (min-width:750px) {
      gap: 20px;
      flex-direction: row;
      align-items: start;

    }

    @include lg-min {
      gap: 60px;
    }

    @include xl-min {
      gap: 100px;
    }
  }

  &__imgWrap {
    margin-bottom: 20px;
    justify-content: center;

    @media (min-width:750px) {
      order: 1;
    }
  }

  &__img {
    max-width: 360px;
    border-radius: 15px;
  }

  &__textBox {
    display: flex;
    flex-direction: column;
    gap: 20px;

    @media (min-width:750px) {
      gap: 40px;
    }
  }

  &__textWrap {
    position: relative;
    max-height: 220px;
    overflow: hidden;

    &:before {
      position: absolute;
      right: 0;
      bottom: 0;
      left: 0;
      height: 100px;
      content: "";
      opacity: 1;
      background: -webkit-gradient(linear, left top, left bottom, from(rgba(255, 255, 255, 0)), to(#fff));
      background: -o-linear-gradient(top, rgba(255, 255, 255, 0) 0, #fff 100%);
      background: linear-gradient(180deg, rgba(255, 255, 255, 0) 0, #fff 100%);
      -webkit-transition: opacity .25s;
      -o-transition: opacity .25s;
      transition: opacity .25s;
    }
  }

  &__desc {
    font-size: $font-size-14;
    line-height: 21px;
    color: $color-black;
    mix-blend-mode: normal;

    @include xl-min {
      // margin-bottom: 30px;
    }
  }

  &__link {
    display: block;
    margin-right: auto;
    font-weight: $font-weight-600;
    font-size: $font-size-14;
    line-height: 14px;
    color: $color-main;
    mix-blend-mode: normal;
    text-decoration: none;

    @include hoverAnimation;

    &:hover {
      color: $color-hover;
    }
  }

  &__linkAlt {
    text-decoration: none;
  }
}