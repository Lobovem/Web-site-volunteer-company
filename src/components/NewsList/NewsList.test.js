import { render, screen } from '@testing-library/react';
import { NewsList } from './NewsList';

describe('NewsList', () => {
  test('renders NewsList component', () => {
    const { container } = render(<NewsList />);
    expect(container).toMatchSnapshot();
  });

  test('search tag h3 (decor)', () => {
    render(<NewsList />);
    const h3Element = screen.getByRole('heading', { level: 3 });
    expect(h3Element).toBeInTheDocument();
  });
});
