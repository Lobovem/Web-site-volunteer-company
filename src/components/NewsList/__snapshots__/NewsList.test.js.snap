// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList renders NewsList component 1`] = `
<div>
  <div
    class="news"
  >
    <h3
      class="news__decor"
    >
      Новини
    </h3>
    <div
      class="news__wrap"
    />
  </div>
</div>
`;
